### PR TITLE
Update the WebIDL parser

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11232)

### CODE REVIEW

File: components/script/dom/webidls/HTMLCollection.webidl [hunk 1]
```diff
@@ -4,7 +4,6 @@
 
 // https://dom.spec.whatwg.org/#interface-htmlcollection
 
-[LegacyUnenumerableNamedProperties]
```
Comment (jdm) [inline components/script/dom/webidls/HTMLCollection.webidl]: Why this change?

Comment (nox) [inline components/script/dom/webidls/HTMLCollection.webidl]: Because I fumbled when I introduced this. They are already unenumerable in HTMLCollection. When I upstreamed it I added (or someone else, I don't remember) a check that looks at ancestor interfaces to see if they are already unenumerable there, in which case derived interfaces shouldn't mention it.

Comment (nox) [inline components/script/dom/webidls/HTMLCollection.webidl]: @Ms2ger made me notice I removed the wrong `LegacyUnenumerableNamedProperties`.
